packages/xml-views-validation/src/validators/attributes/unknown-attribute-key.ts

```diff
@@ -0,0 +1,179 @@
+import { includes, find } from "lodash";
+import { XMLAttribute } from "@xml-tools/ast";
+import {
+  UI5SemanticModel,
+  UI5Aggregation,
+  UI5Association,
+  UI5Event,
+  UI5Prop,
+} from "@ui5-language-assistant/semantic-model-types";
+import {
+  getUI5ClassByXMLElement,
+  getUI5AggregationByXMLElement,
+  isXMLNamespaceKey,
+  flattenProperties,
+  flattenEvents,
+  flattenAssociations,
+  flattenAggregations,
+  ui5NodeToFQN,
+} from "@ui5-language-assistant/logic-utils";
+import { UnknownAttributeKeyIssue } from "../../../api";
+import { TEMPLATING_NS, CUSTOM_DATA_NS } from "../../utils/special-namespaces";
+
+export function validateUnknownAttributeKey(
+  attribute: XMLAttribute,
+  model: UI5SemanticModel
+): UnknownAttributeKeyIssue[] {
+  if (attribute.syntax.key === undefined || attribute.key === null) {
+    // Can't give an error without a position or value
+    return [];
+  }
+
+  if (isAttributeNameAlwaysValid(attribute)) {
+    return [];
+  }
+
+  // Only validate (known) classes and aggregations
+  const xmlElement = attribute.parent;
+  const ui5Class = getUI5ClassByXMLElement(xmlElement, model);
+  let ui5Aggregation: UI5Aggregation | undefined = undefined;
+  if (ui5Class === undefined) {
+    ui5Aggregation = getUI5AggregationByXMLElement(xmlElement, model);
+  }
+
+  // Aggregations don't have additional allowed attributes
+  if (ui5Aggregation !== undefined) {
+    return [
+      {
+        kind: "UnknownAttributeKey",
+        message: `Unknown attribute key: ${attribute.key}`,
+        offsetRange: {
+          start: attribute.syntax.key.startOffset,
+          end: attribute.syntax.key.endOffset,
+        },
+        severity: "error",
+      },
+    ];
+  }
+
+  if (ui5Class === undefined) {
+    return [];
+  }
+
+  // Special case: "class" property can be used on all Controls and selected Elements.
+  // TODO Should this be in the model? Should it be suggested for code completion?
+  if (attribute.key === "class") {
+    return [];
+  }
+
+  // Special case: "binding" property is a synonym for "objectBindings" special setting on ManagedObject
+  // TODO Should this be in the model? Should it be suggested for code completion?
+  if (attribute.key === "binding") {
```

Comment:
naively sounds like the model needs to be expended

Reply:
I asked Yuval, waiting for his answer

Reply:
As it should not be offered in code completion I prefer to keep it in the validation, at least for now